scripts/complete-exchange.ts

```diff
@@ -0,0 +1,174 @@
+import { MetaTxConfig } from "@bosonprotocol/common";
+import fs from "fs";
+import { resolve } from "path";
+import { program } from "commander";
+import { Wallet, providers } from "ethers";
+import {
+  getDefaultConfig,
+  EnvironmentType,
+  CoreSDK
+} from "../packages/core-sdk/src";
+import { ExchangeState } from "../packages/core-sdk/src/subgraph";
+import { EthersAdapter } from "../packages/ethers-sdk/src";
+
+// TODO: read real value in the diamond (IBosonConfigHandler::getMaxExchangesPerBatch())
+const MAX_EXCHANGES_PER_BATCH = 140;
+
+program
+  .description("Complete one or several exchanges.")
+  .argument(
+    "<PRIVATE_KEY>",
+    "Private key of the account issuing the transaction."
+  )
+  .option("-e, --env <ENV_NAME>", "Target environment", "testing")
+  .option(
+    "-m, --metaTx <META_TX_CONFIG>",
+    "JSON file with the meta tx parameters"
+  )
+  .option(
+    "--exchanges <EXCHANGES>",
+    "Comma separated list of exchanges to complete"
+  )
+  .option(
+    "--sellerId <SELLER_ID>",
+    "Id of the seller to complete all completable exchanges for"
+  )
+  .option("--dryRun", "do not send the transaction")
+  .parse(process.argv);
+
+async function main() {
+  const [privateKey] = program.args;
+
+  const opts = program.opts();
+  const envName = (opts.env as EnvironmentType) || "testing";
+  const defaultConfig = getDefaultConfig(envName);
+  const wallet = new Wallet(privateKey);
+  let metaTx: Partial<MetaTxConfig> | undefined = undefined;
+  if (opts.metaTx) {
+    //
+    // JSON file example:
+    // {
+    //   "apiKey": "<BICONOMY_API_KEY>>",
+    //   "apiIds": {
+    //     "<DIAMOND_ADDRESS>": {
+    //       "executeMetaTransaction": "<BICONOMY_API_ID>"
+    //     }
+    //   }
+    // }
+    // Where:
+    //  - <BICONOMY_API_KEY>: is the ApiKEy of the Biconomy Project
+    //  - <DIAMOND_ADDRESS>: in the Boson Protocol diamond contract address in the targeted environment
+    //  - <BICONOMY_API_ID> : is the ApiId for Boson Diamond "executeMetaTransaction" method
+    //
+    const rawMetaTx = fs.readFileSync(resolve(__dirname, opts.metaTx));
+    metaTx = JSON.parse(rawMetaTx.toString());
+  }
+  const coreSDK = CoreSDK.fromDefaultConfig({
+    web3Lib: new EthersAdapter(
+      new providers.JsonRpcProvider(defaultConfig.jsonRpcUrl),
+      wallet
+    ),
+    envName,
+    metaTx
+  });
+  if (opts.metaTx && !coreSDK.isMetaTxConfigSet) {
+    throw `Invalid metaTx configuration:\n${JSON.stringify(metaTx)}`;
+  }
+
+  const exchangeIds = opts.exchanges
+    ? (opts.exchanges as string).split(",")
+    : [""];
+
+  const exchanges = await coreSDK.getExchanges({
+    exchangesFilter: {
+      state_in: [ExchangeState.Redeemed],
+      id_in: exchangeIds
+    }
+  });
+  if (opts.sellerId) {
+    exchanges.push(
+      ...(await coreSDK.getExchanges({
+        exchangesFirst: 1000,
```

Comment:
I guess that nobody is going to call this script with more than 1000 exchanges id so this should be fine